src/dune_rules/lock_rules.ml

```diff
@@ -409,12 +408,60 @@ let copy_lock_dir ~target ~lock_dir ~deps ~files =
             ~dirs:(Path.Build.Set.singleton target))
 ;;
 
-let setup_copy_rules ~dir:target ~lock_dir =
-  let+ deps, files = Source_deps.files (Path.source lock_dir) in
+let files dir =
+  let rec recurse dir =
```

Comment:
Does the previous code work? I don't see any reason it shouldn't here. IIUC we are now using external so the build context will not be dropped which `Source_deps.files` does.

Reply:
No, `Source_deps.files` returns `(Dep.Set.empty, Path.Set.empty)` as it uses `Source_tree.find_dir` which does (correctly) return `None` on that path.